cmd/influx_inspect/deletetsm/deletetsm.go

```diff
@@ -0,0 +1,161 @@
+// Package deletetsm bulk deletes a measurement from a raw tsm file.
+package deletetsm
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/engine/tsm1"
+)
+
+// Command represents the program execution for "influxd deletetsm".
+type Command struct {
+	// Standard input/output, overridden for testing.
+	Stderr io.Writer
+	Stdout io.Writer
+
+	measurement string // measurement to delete
+	verbose     bool   // verbose logging
+}
+
+// NewCommand returns a new instance of Command.
+func NewCommand() *Command {
+	return &Command{
+		Stderr: os.Stderr,
+		Stdout: os.Stdout,
+	}
+}
+
+// Run executes the command.
+func (cmd *Command) Run(args ...string) (err error) {
+	fs := flag.NewFlagSet("deletetsm", flag.ExitOnError)
+	fs.StringVar(&cmd.measurement, "measurement", "", "")
+	fs.BoolVar(&cmd.verbose, "v", false, "")
+	fs.SetOutput(cmd.Stdout)
+	fs.Usage = cmd.printUsage
+	if err := fs.Parse(args); err != nil {
+		return err
+	} else if fs.NArg() == 0 {
+		fmt.Printf("path required\n\n")
+		fs.Usage()
+		return nil
+	}
+
+	if !cmd.verbose {
+		log.SetOutput(ioutil.Discard)
+	}
+
+	// Validate measurement flag.
+	if cmd.measurement == "" {
+		return fmt.Errorf("measurement name required")
+	}
+
+	// Process each TSM file.
+	for _, path := range fs.Args() {
+		log.Printf("processing: %s", path)
+		if err := cmd.process(path); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cmd *Command) process(path string) error {
+	// Open TSM reader.
+	input, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer input.Close()
+
+	r, err := tsm1.NewTSMReader(input)
+	if err != nil {
+		return fmt.Errorf("unable to read %s: %s", path, err)
+	}
+	defer r.Close()
+
+	// Remove previous temporary files.
+	outputPath := path + ".rewriting"
```

Comment:
I'd use `.tmp` so that the server will remove any failed runs.  `.rewriting` will linger forever if the process crashes and is not retried.

Reply:
Fixed in 8c9d179.